tests/models/speech_to_text/test_modeling_speech_to_text.py

```diff
@@ -284,6 +284,16 @@ class Speech2TextModelTest(ModelTesterMixin, GenerationTesterMixin, PipelineTest
 
     input_name = "input_features"
 
+    def _get_input_ids_and_config(self, batch_size=2):
+        config, input_ids, attention_mask, max_length = GenerationTesterMixin._get_input_ids_and_config(self)
+
+        # overwrite because otherwise the attention mask'd have the same shape than the inputs here, which is a 3D tensor.
```

Comment:
if not too much work, might be nice to explain why it is 3D (it seems this is special situation to this model?) and why we need 2D.